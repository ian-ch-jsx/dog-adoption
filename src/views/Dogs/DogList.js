import { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { getDogs } from '../../services/dogs';
import DogCard from '../../components/Dog/DogCard';
import './Dogs.css';

export default function DogList() {
  const [dogs, setDogs] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const data = await getDogs();
      setDogs(data);
      setLoading(false);
    };
    fetchData();
  }, []);
  if (loading) return <h1>loading...</h1>;
  return (
    <div className="dog-container">
      {dogs.map((dog) => (
        <Link to={`/dogs/${dog.id}`} key={dog.id}>
          <DogCard {...dog} />
        </Link>
      ))}
    </div>
  );
}
