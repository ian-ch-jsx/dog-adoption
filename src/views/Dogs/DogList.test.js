import { render } from '@testing-library/react';
import DogList from './DogList';

test('renders dog info', () => {
  const { container } = render(
    <DogList
      name="Barton"
      image="https://placedog.net/500?id=6"
      age={1}
      breed="Weimaraner"
      bio="dog is cute"
    />
  );
  expect(container).toMatchSnapshot();
});
