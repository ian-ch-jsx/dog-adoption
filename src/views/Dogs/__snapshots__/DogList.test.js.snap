// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dog info 1`] = `
<div>
  <h1>
    loading...
  </h1>
</div>
`;
